sqlmesh/integrations/dlt.py

```diff
@@ -0,0 +1,169 @@
+import typing as t
+import click
+from pydantic import ValidationError
+from sqlglot import exp
+from sqlmesh.core.config.connection import parse_connection_config
+
+
+def generate_dlt_models_and_settings(
+    pipeline_name: str, dialect: str
+) -> t.Tuple[t.Set[t.Tuple[str, str]], str]:
+    """This function attaches to a DLT pipeline and retrieves the connection configs and
+    SQLMesh models based on the tables present in the pipeline's default schema.
+    """
+
+    import dlt
+    from dlt.common.schema.utils import has_table_seen_data, is_complete_column
+    from dlt.pipeline.exceptions import CannotRestorePipelineException
+
+    try:
+        pipeline = dlt.attach(pipeline_name=pipeline_name)
+    except CannotRestorePipelineException:
+        raise click.ClickException("Could not attach to pipeline {pipeline_name}")
```

Comment:
Missing `f""`. The string won't be interpolated

Reply:
Thanks my bad missed this